common/persistence/nosql/nosqlplugin/cassandra/gocql/session.go

```diff
@@ -108,14 +114,20 @@ func (s *session) refresh() {
 }
 
 func initSession(
+	ctx context.Context,
 	config config.Cassandra,
 	resolver resolver.ServiceResolver,
+	tracerProvider trace.TracerProvider,
 ) (*gocql.Session, error) {
 	cluster, err := NewCassandraCluster(config, resolver)
 	if err != nil {
 		return nil, err
 	}
-	return cluster.CreateSession()
+	return otelgocql.NewSessionWithTracing(
+		ctx,
```

Comment:
Why not to use `context.Background()` here? I see that you want to pass it from caller but `refresh()` still has no other option besides using `context.Background()`. Does it really worth to change `NewSession` params list?

Reply:
The CreateSession/NewSession calls actually perform the Cassandra connection setup/teardown (unlike say grpc.Dial which does its work in the background by default). This context (and the parent span it may contain) is used emit spans and metrics with connection information.